server/routes/academy.js

```diff
@@ -1256,26 +1256,35 @@ router.post(
       });
     }
 
-    let classesQuery = await network.query(networkObj, 'GetClassesByTeacher', username);
+    let classesQuery = await network.query(networkObj, 'GetClass', classId);
     if (!classesQuery.success) {
       return res.status(500).json({
         success: false,
-        msg: "Can't query classes by teacher"
+        msg: "Can't query class in chaincode!"
       });
     }
 
-    let listClasses = JSON.parse(classesQuery.msg);
+    classesQuery = JSON.parse(classesQuery.msg);
 
-    if (listClasses) {
-      for (let index = 0; index < listClasses.length; index++) {
-        const element = listClasses[index];
-        if (element.ClassID === classId) {
-          return res.status(500).json({
-            success: false,
-            msg: 'The class has been added!'
-          });
-        }
+    if (classesQuery.TeacherUsername !== '') {
+      if (classesQuery.TeacherUsername === username) {
```

Comment:
sao lại 2 vòng if lồng nhau thế này, chỉ cần kiểm tra điều kiện `if (classesQuery.TeacherUsername === username) {` thôi chứ, còn `username` phải validate từ lúc mới vào api rồi

Reply:
dạ còn điều kiện ở dưới nữa ạ, nếu class.TeacherUsername == cái username  truyền vào thì return msg: 'The class was added for this teacher!', còn nếu khác thì return `The class was added for teacher - ${classesQuery.TeacherUsername}!`

Reply:
ok (y)